pep-0012.rst

```diff
@@ -438,43 +434,92 @@ Hyperlinks
 ----------
 
 When referencing an external web page in the body of a PEP, you should
-include the title of the page in the text, with either an inline
-hyperlink reference to the URL or a footnote reference (see
-`Footnotes`_ below).  Do not include the URL in the body text of the
-PEP.
+include the title of the page or a suitable description in the text, with
+either an inline hyperlink or a separate explicit target with the URL.
+Do not include bare URLs in the body text of the PEP, and use HTTPS
+links wherever available.
 
 Hyperlink references use backquotes and a trailing underscore to mark
 up the reference text; backquotes are optional if the reference text
-is a single word.  For example::
+is a single word.  For example, to reference a hyperlink target named
+``Python web site``, you would write:
+
+.. code-block:: rst
+
+    In this paragraph, we refer to the `Python web site`_.
+
+which renders as:
 
     In this paragraph, we refer to the `Python web site`_.
 
-An explicit target provides the URL.  Put targets in a References
-section at the end of the PEP, or immediately after the reference.
+.. _Python web site: https://www.python.org/
+
+If you intend to only reference a link once, and want to define it inline
+with the text, insert the link into angle brackets (``<>``) after the text
+you want to link, but before the closing backtick, with a space between the
+text and the opening backtick. You should also use a double-underscore after
+the closing backtick instead of a single one, which makes it an anonymous
+reference to avoid conflicting with other target names. For example:
```

Comment:
Is it worth giving so much detail on how to write a link or to link to the reST docs at https://docutils.sourceforge.io/docs/user/rst/quickref.html#indirect-hyperlink-targets and provide the examples?

Same below for the other link styles.

Reply:
My initial thought as well was just to point to the docutils reference or similar for the details, but:

* The other sections all explicitly describe how to use their syntax elements without reference to docutils, so we should be consistent (might be worth revising, but out of scope here)
* The docutils quickref doesn't actually explicitly discuss or provide examples for several of the things discussed here and commonly used/misused in PEPs, including indirect references of the form used here, multi-word link targets or the necessity of the space between the text and `<`, while the docutils full reference has a much higher level of comprehensive detail, much of which is not relevant here
* It contains a number of PEP-specific recommendations and guidance, such as where to put link targets, where and why to prefer different forms, and when to use anonymous references; the examples are tailored to PEPs; and the section is carefully organized to structure the key information in a logical and accessible manner for a PEP writer, rather than scattered all about.

So, my preference would be to keep it as-is, following the general status quo, and consider offloading the RST syntax details more generally in another PR. But if its overly lengthy, we could slim down some bits, like the rendered form for the different ways of specifying links.

Reply:
The detailed comments are from a time when we had to teach people how to write reST for a PEP since PEPs predate the format. Since reST is now so old, I don't think it's important long-term.

Reply:
I see, thanks for the historical background @brettcannon . I agree it would likely be a good idea to slim down the detailed reST syntax descriptions in PEP 12 to just a brief summary and/or any PEP-specific guidance, and link to the relevant docutils/Sphinx guides and documentation for the rest, rather than belaboring the PEP with such details that are described elsewhere.

That said, that's a distinct and more expansive change than just updating the existing descriptions of how to use hyperlinks and footnotes (as is the scope of #2130 that this resolves) and something we should probably discuss in a separate issue first to make sure we're all in agreement and decide on a consistent approach to this throughout the PEP before going ahead with it.

For now, I've reverted the addition of all the "renders as" example blocks and their associated text for the hyperlinks section, along with an extra paragraph and the explanation and examples of numbered footnotes. I suggest we go ahead and merge this with those changes for now, so the PEP accurately reflects the current recommendations for links/references as agreed on in #2130 (and so I can move ahead with #2266), and I can open a new issue for us to discuss and decide upon the approach to take with PEP 12 moving forward, and implement that once consensus is reached. Does that sound good to you?

Reply:
SGTM

Reply:
Thanks! Opened as #2337